src/Transactions/TransactionDetails/TransactionDetailsContainer.js

```diff
@@ -40,42 +48,62 @@ const TransactionDetailsContainer = ({
 
   useEffect(
     () => {
-      mutator.transactionFunds.reset();
-      mutator.transactionDetails.reset();
-      mutator.fiscalYear.reset();
+      setTransaction();
+      setTransactionFunds();
+      setFiscalYear();
+      setExpenseClass();
+
       mutator.transactionDetails.GET()
-        .then(({ fiscalYearId, fromFundId, toFundId }) => {
-          mutator.fiscalYearId.replace(fiscalYearId);
-          mutator.fiscalYear.GET();
+        .then(transactionResponse => {
+          setTransaction(transactionResponse);
 
-          const fundsQuery = [fromFundId, toFundId].map(id => `id == ${id}`).join(' OR ');
+          const { fromFundId, toFundId, fiscalYearId, expenseClassId } = transactionResponse;
+          const funds = [fromFundId, toFundId].filter(Boolean);
+          const fundsQuery = funds.map(id => `id == ${id}`).join(' OR ');
 
-          mutator.transactionFunds.GET({ params: { query: fundsQuery } });
-        });
+          const fiscalYearPromise = mutator.fiscalYear.GET({
+            path: `${FISCAL_YEARS_API}/${fiscalYearId}`,
+          });
+          const transactionFundsPromise = mutator.transactionFunds.GET({ params: { query: fundsQuery } });
+          const expenseClassPromise = expenseClassId
+            ? (
+              mutator.transactionExpenseClass.GET({
```

Comment:
I don't think we choose that way to display data. Usually we leverage fetching code to separate components, to not make "fat" controllers
https://github.com/folio-org/stripes-acq-components/blob/master/lib/FieldOrganization/FieldOrganization.js
https://github.com/folio-org/stripes-acq-components/blob/master/lib/Organization/OrganizationValue/OrganizationValue.js
https://github.com/folio-org/stripes-smart-components/blob/master/lib/UserName/UserName.js